app/pages/event1/employer/LoanDetailsPage.scala

```diff
@@ -30,4 +32,11 @@ case object LoanDetailsPage extends QuestionPage[LoanDetails] {
 
   override def route(waypoints: Waypoints): Call =
     routes.LoanDetailsController.onPageLoad(waypoints)
+
+  override protected def nextPageNormalMode(waypoints: Waypoints, answers: UserAnswers): Page = {
+    answers.get(LoanDetailsPage) match {
+      case Some(_) => PaymentValueAndDatePage
+      case _ => IndexPage
```

Comment:
fields are optional. So we are going to paymentValueAndDate page anyway, aren't we ?
It seems this can be enough.
```
  override protected def nextPageNormalMode(waypoints: Waypoints, answers: UserAnswers): Page = {
    PaymentValueAndDatePage
  }
```

Reply:
Resolved in next commit